### PR TITLE
Require SSL connection to database if MariaDB SSL is configured

### PR DESCRIPTION
**Pull request by jsuchome**
With this settings, non-SSL access is forbidden for the affected
users.

This is based on @s-t-e-v-e-n-k patch. 

**NOTE**: 
 - db_maker access does not work for insecure connection, requires this fix in rubygem-mysql2:
 https://github.com/brianmario/mysql2/pull/889
 - currently present in our rubygem package

### CODE REVIEW

File: chef/cookbooks/database/libraries/provider_database_mysql_user.rb [hunk 1]
```diff
@@ -111,13 +111,21 @@ def user_present?(username, host)
...
             username: new_resource.connection[:username],
             password: new_resource.connection[:password],
             port: new_resource.connection[:port]
-          )
+          }
+          if new_resource.connection[:ssl][:enabled]
+            if new_resource.connection[:ssl][:insecure]
+              client_options[:sslverify] = false
+            else
+              client_options[:sslca] = new_resource.connection[:ssl][:ca_certs]
```
[the remark this answers is not in the document]
Comment (jsuchome) [reply chef/cookbooks/database/libraries/provider_database_mysql_user.rb]: Ah, true indeed

**Comment by jsuchome** [comment]
Rebased (no code change)

**Comment by jsuchome** [comment]
> This seems to be missing keystone at least.

Added

**Comment by jsuchome** [comment]
@rhafer as your patch is currently living in build service, this PR could be merged actually.

**Comment by jsuchome** [comment]
Current blocker is neutron behavior, see bug https://bugzilla.suse.com/show_bug.cgi?id=1060405

**Comment by rsalevsky** [comment]
@jsuchome Would splitting the neutron part in a separate PR help unblocking this PR?

**Comment by jsuchome** [comment]
@rsalevsky ok, I've removed neutron part in last commit ... but I'm not sure how severe the problem is, maybe we're gonna have to rework it again

File: chef/cookbooks/horizon/recipes/server.rb [hunk 2]
```diff
@@ -333,6 +333,7 @@
     only_if { !ha_enabled || CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
 
+database_ssl = db_settings[:connection][:ssl][:enabled] && !db_settings[:connection][:ssl][:insecure]
```
Comment (rhafer) [inline chef/cookbooks/horizon/recipes/server.rb]: Please add a comment with an explanation why we're not enabling ssl for horizon in "insecure" mode.

Comment (jsuchome) [reply chef/cookbooks/horizon/recipes/server.rb]: Done

File: chef/cookbooks/horizon/recipes/server.rb [hunk 3]
```diff
@@ -333,6 +333,9 @@
     only_if { !ha_enabled || CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
 
+# We do not require SSL connectopn for horizon user in case of insecure DB setup,
+# because horizon's django uses a library (MySQLdb) that does not support insecure connection.
+database_ssl = db_settings[:connection][:ssl][:enabled] && !db_settings[:connection][:ssl][:insecure]
```
[the remark this answers is not in the document]
Comment (rhafer) [reply chef/cookbooks/horizon/recipes/server.rb]: Would you mind addressing this as well?

**Pull request merged**